### PR TITLE
Update to 1.19.3

### PR DESCRIPTION
**Pull request by frej4189**
Depends on:

- https://github.com/PrismarineJS/node-minecraft-data/pull/253 for properly parsing the updated minecraft-data
- https://github.com/PrismarineJS/minecraft-data/pull/672
- https://github.com/PrismarineJS/minecraft-data/pull/673

- [x] Client chat validation
- [x] Client chat sending
- [x] Server chat validation
- [x] Update documentation
- [x] Fix tests

Code might also need some tidying up, but the functionality is there

### CODE REVIEW

**Comment by rom1504** [comment]
noticed we were not testing .2 and .3, fixed that

**Comment by rom1504** [comment]
@frej4189 can you resolve the merge conflict ?

**Comment by rom1504** [comment]
1.19.2 and .3 are failing, please fix

File: test/serverTest.js [hunk 1]
```diff
@@ -83,7 +84,10 @@ for (const supportedVersion of mc.supportedVersions) {
         senderTeam: undefined,
         timestamp: Date.now(),
         salt: 0n,
-        signature: Buffer.alloc(0)
+        signature: mcData.supportFeature('useChatSessions') ? undefined : Buffer.alloc(0),
```
Comment (rom1504) [inline test/serverTest.js:87]: useChatSessions does not exist

Comment (rom1504) [reply test/serverTest.js:87]: nevermind

Comment (frej4189) [reply test/serverTest.js:87]: https://github.com/PrismarineJS/minecraft-data/blob/1f1eec2ba642915aca4fec76e8bbe086bd6d73b6/data/pc/common/features.json#L691

Comment (rom1504) [reply test/serverTest.js:87]: why is signature undefined if it exists ?

Comment (frej4189) [reply test/serverTest.js:87]: If we are in 1.19.3 we don't want to send the signature field

File: test/packetTest.js [hunk 2]
```diff
@@ -215,6 +228,7 @@ function getValue (_type, packet) {
   }
 }
 
+mc.supportedVersions = ['1.19.3']
```
Comment (rom1504) [inline test/packetTest.js]: seems like your forgot this

you can use -g of mocha to run only some tests

Comment (frej4189) [reply test/packetTest.js]: Good to know. Have already reverted it

File: docs/API.md [hunk 3]
```diff
@@ -212,6 +212,10 @@ The user's username.
 
 The user's session, as returned by the Yggdrasil system. (only client-side)
 
+### client._session
+
+The user's chat session. Only in 1.19.3 and above. You should not need to access this.
```
Comment (extremeheat) [inline docs/API.md:217]: Underscore shouldn't be documented in stable API

File: src/client/setProtocol.js [hunk 4]
```diff
@@ -38,7 +37,7 @@ module.exports = function (client, options) {
                 : client.profileKeys.signature
             }
           : null,
-        playerUUID: client.session?.selectedProfile?.id
+        playerUUID: !mcData.supportFeature('useChatSessions') ? client.session?.selectedProfile?.id : undefined
```
Comment (extremeheat) [inline src/client/setProtocol.js:40]: Protodef will ignore fields if not applicable to version, version checks on them not needed

Comment (frej4189) [reply src/client/setProtocol.js:40]: I got errors without this. Might have been because of the MC data issues earlier

File: docs/API.md (hunk 3 above)
Comment (frej4189) [reply docs/API.md:217]: Are there any unstable API docs in NMP?

Comment (extremeheat) [reply docs/API.md:217]: The API.md is the stable API

Comment (frej4189) [reply docs/API.md:217]: Sure. But do we have anywhere to document unstable API currently? Or should we just not document this?

Comment (extremeheat) [reply docs/API.md:217]: Yeah you can remove this

**Comment by rom1504** [comment]
Is this ready? Looks like it

**Comment by frej4189** [comment]
> Is this ready? Looks like it

Believe so

**Review by Prosciuttino — APPROVED** [review]

**Pull request merged**